core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2793,6 +2799,9 @@ protected Scope visitMerge(Merge merge, Optional<Scope> scope)
             Relation relation = merge.getTarget();
             Table table = getMergeTargetTable(relation);
             QualifiedObjectName tableName = createQualifiedObjectName(session, table, table.getName());
+            if (metadata.isMaterializedView(session, tableName)) {
+                throw semanticException(NOT_SUPPORTED, merge, "MERGE INTO a materialized view is not supported");
```

Comment:
nit: INTO here reads as SQL syntax clause and preposition both : ) how about 

MERGE INTO is not supported for a materialized view. Or 
merging into a materialized view is not supported.